infra/modules/worklytics-connector-specs/main.tf

```diff
@@ -247,6 +247,66 @@ locals {
         "/v1.0/groups",
         "/v1.0/groups/{group-id}/members"
       ]
+    },
+    "msft-teams" : {
+      source_kind : "msft-teams"
+      worklytics_connector_id : "msft-teams-psoxy",
+      display_name : "Microsoft Teams"
+      identifier_scope_id : "azure-ad"
+      source_auth_strategy : "oauth2_refresh_token"
+      target_host : "graph.microsoft.com"
+      required_oauth2_permission_scopes : [],
+      required_app_roles : [
+      # Least privileged permissions
+      "Team.ReadBasic.All",
+      "Channel.ReadBasic.All",
+      "Chat.ReadBasic.All",
+      "Chat.Read.All",
+      #"ChannelMessage.Read.Group*",
+      "ChannelMessage.Read.All",
+      "CallRecords.Read.All",
+      "OnlineMeetings.Read.All"
+
+      # Higher privileged permissions
+#        "TeamSettings.Read.All",
+#        "TeamSettings.ReadWrite.All",
+#        "ChannelSettings.Read.All",
+#        "ChannelSettings.ReadWrite.All",
+#        "Chat.ReadWrite.All",
+#        "ChatMessage.Read.Chat",
+#        "Calls.AccessMedia.All",
+#        "OnlineMeetings.ReadWrite.All",
+#        "OnlineMeetingArtifact.Read.All"
+      ],
+      environment_variables : {
+        GRANT_TYPE : "workload_identity_federation"
+        # by default, assumed to be of type 'urn:ietf:params:oauth:client-assertion-type:jwt-bearer'
+        TOKEN_SCOPE : "https://graph.microsoft.com/.default"
+        REFRESH_ENDPOINT : "https://login.microsoftonline.com/${var.msft_tenant_id}/oauth2/v2.0/token"
+      }
+      example_api_calls : [
+        "/beta/{apiVersion}/teams",
+        "/beta/{apiVersion}/teams/{teamId}/allChannels",
+        "/beta/{apiVersion}/users/{userId}/chats",
+        "/beta/{apiVersion}/teams/{teamId}/channels/{channelId}/messages",
+        "/beta/{apiVersion}/teams/{teamId}/channels/{channelId}/messages/delta",
+        "/beta/{apiVersion}/chats/{chatId}/messages",
+        "/beta/{apiVersion}/communications/calls/{callId}",
+        "/beta/{apiVersion}/communications/callRecords/{callChainId}",
+        "/beta/{apiVersion}/users/{userId}/onlineMeetings",
+
+        "/v1.0/{apiVersion}/teams",
+        "/v1.0/{apiVersion}/teams/{teamId}/allChannels",
+        "/v1.0/{apiVersion}/users/{userId}/chats",
+        "/v1.0/{apiVersion}/teams/{teamId}/channels/{channelId}/messages",
+        "/v1.0/{apiVersion}/teams/{teamId}/channels/{channelId}/messages/delta",
+        "/v1.0/{apiVersion}/chats/{chatId}/messages",
+        "/v1.0/{apiVersion}/communications/calls/{callId}",
+        "/v1.0/{apiVersion}/communications/callRecords/{callChainId}",
+        "/v1.0/communications/callRecords/getDirectRoutingCalls(fromDateTime=2019-11-01,toDateTime=2019-12-01)",
+        "/v1.0/communications/callRecords/getPstnCalls(fromDateTime=2019-11-01,toDateTime=2019-12-01)",
```

Comment:
@aperez-worklytics  do we need to mentioned this in `external_token_todo` section ? I don't see this for saleforce, (except this one: _Finally, we recommend to run `test-salesforce` script with all the queries in the example to ensure the expected information covered by rules _).

Reply:
No, this is an example call and only for testing purposes. If we put the date fixed the endpoint will fail -and that could cause *false positive failures*-. Using the formula we ensure that we are inside the specs, so in case someone test it, it will not fail due an invalid default parameters.